src/scipp/io/hdf5.py

```diff
@@ -344,14 +347,16 @@ def _write_mapping(
 
 
 def _read_mapping(
-    group: h5.Group, override: Mapping[str, h5.Group] | None = None
+    group: h5.Group,
+    override: Mapping[str, h5.Group] | None = None,
+    aligned: bool | None = None,
 ) -> VariableLike:
     if override is None:
         override = {}
     return {
         g.attrs['name']: override[g.attrs['name']]
         if g.attrs['name'] in override
-        else _HDF5IO.read(g)
+        else _HDF5IO.read(g, aligned=aligned)
```

Comment:
This does not match the common interface of readers. And it doesn't make sense in general because only coords can be aligned, not, e.g., data groups.

I think it would be easiest to modify `_read_legacy_attrs_into` to add coords to a `DataArray` directly instead of the `contents` dict. Then you can use `Coords.set_aligned`.